libs/dh/market-participant/actors/feature-actors/src/lib/edit/dh-actors-edit-actor-modal.component.ts

```diff
@@ -67,15 +67,18 @@ import { WattPhoneFieldComponent } from '@energinet-datahub/watt/phone-field';
   ],
 })
 export class DhActorsEditActorModalComponent {
-  private readonly dataAccessStore = inject(DhMarketParticipantActorsEditActorDataAccessApiStore);
   private readonly formBuilder = inject(FormBuilder);
   private readonly transloco = inject(TranslocoService);
   private readonly toastService = inject(WattToastService);
 
-  @ViewChild(WattModalComponent)
-  innerModal: WattModalComponent | undefined;
+  innerModal = viewChild.required<WattModalComponent>(WattModalComponent);
 
-  @Input() actor: DhActorExtended | undefined;
+  actor = input<DhActorExtended>();
```

Comment:
question: is it possible to make this required by wrapping the modal in an `@if`? something like this:

```html
@if (actor(); as actor)
  <dh-actors-edit-actor-modal #editActorModal [actor]="actor" />
}
```

Or is it intentionally that way to avoid instantiating the modal multiple times? 🙂

Reply:
No, because then the stupid #editActorModal ref on the button wont work :-(

Reply:
Ah I see 😄